core/templates/dev/head/app.js

```diff
@@ -1220,3 +1220,20 @@ oppia.factory('codeNormalizationService', [function() {
     }
   };
 }]);
+
+// Service for retrieving algorithm id of classifier for a given interaction id.
+oppia.factory('algorithmIdRetrievalService', [function() {
```

Comment:
@AllanYangZhou In this case, we just want the algorithm_id. This is required because the algorithm_id is what we use to retrieve its corresponding service. Basically speaking, there is a one-to-one correspondence between algorithm_id and prediction services.